sled-agent/src/params.rs

```diff
@@ -374,59 +389,60 @@ impl crate::smf_helper::Service for ServiceType {
     }
 }
 
-impl From<ServiceType> for sled_agent_client::types::ServiceType {
-    fn from(s: ServiceType) -> Self {
+/// Error returned by attempting to convert an internal service (i.e., a service
+/// started autonomously by sled-agent) into a
+/// `sled_agent_client::types::ServiceType` to be sent to a remote sled-agent.
+#[derive(Debug, Clone, Copy, Error)]
+#[error("This service may not be requested externally")]
+pub struct InternalServiceOnlyError;
```

Comment:
Looking over this again, this may be a confusing name given "internal / external" in other contexts. This is supposed to mean "internal to sled agent". This is the error type returned if we try to serialize one of the `serde::skip`'d services into a `sled_agent_client::types::ServiceType`; I'm very open to better naming ideas. Maybe `AutonomousServiceOnlyError`, or even `SwitchZoneServiceError` (although that's technically a lie because we also autonomously start Maghemite in the global zone).

Reply:
I like `AutonomousServiceOnlyError` - the distinction really is "who controls the deployment of this service", so that's intuitive.